src/error.rs

```diff
@@ -995,6 +997,29 @@ impl ToDiagnostic<FileId> for EvalError {
                     .with_labels(labels)
                     .with_notes(vec![String::from(INTERNAL_ERROR_MSG)])]
             }
+            EvalError::EqError { eq_pos, term: t } => {
+                let label = format!(
+                    "an argument has type {}, which cannot be compared for equality",
+                    t.term
+                        .type_of()
+                        .unwrap_or_else(|| String::from("<unevaluated>")),
```

Comment:
Hm, I just copied this over from the output for `TypeError` without thinking too hard about it. I agree that it's a little odd to deal with a case that should be impossible here, but I also don't know that the `EqError` diagnostics should be making assumptions about how the `==` operation itself is implemented. Which is to say that since `type_of` returns an `Option` I think it's reasonable to handle it this way rather than asserting.